hack/copyright.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	// license compatible for Go and Proto files.
+	license = []byte(`/*
+Copyright 2020 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+`)
+
+	// since different year in Copyright, just check this exist or not.
+	licenseFlag = []byte(`http://www.apache.org/licenses/LICENSE-2.0`)
+
+	usage = []byte(`
+Usage:
+  check:
+  	go run copyright.go check
+  update:
+	  go run copyright.go
+`)
+)
+
+func checkAndApplyLicense(update bool) error {
+	return filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Filter out stuff that does not need copyright.
+		if info.IsDir() {
+			switch path {
+			case "vendor":
+				return filepath.SkipDir
+			}
+			return nil
+		}
+		if strings.HasSuffix(path, ".pb.go") {
+			return nil
+		}
+		if filepath.Ext(path) != ".proto" && filepath.Ext(path) != ".go" {
+			return nil
+		}
+
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		if !strings.Contains(string(b), string(licenseFlag)) {
```

Comment:
The checker is to ensure if inputed files have copyright header, relying on "http://www.apache.org/licenses/LICENSE-2.0" as keywords would break easily if any none APL license file inputed.

Reply:
It check our own files, which should always use apache v2 license.

Reply:
Shall we consider checking the `sh` file ? @daixiang0 @kevin-wangzefeng @fisherxu

Reply:
Done.